llama-index-core/llama_index/core/postprocessor/rankGPT_rerank.py

```diff
@@ -86,6 +86,50 @@ def _postprocess_nodes(
         else:
             return nodes[: self.top_n]
 
+    async def _async_postprocess_nodes(
+        self,
+        nodes: List[NodeWithScore],
+        query_bundle: Optional[QueryBundle] = None,
+    ) -> List[NodeWithScore]:
+        items = {
+            "query": query_bundle.query_str,
+            "hits": [{"content": node.get_content()} for node in nodes],
+        }
+
+        messages = self.create_permutation_instruction(item=items)
+        permutation = await self.async_run_llm(messages=messages)
+        if permutation.message is not None and permutation.message.content is not None:
+            rerank_ranks = self._receive_permutation(
+                items, str(permutation.message.content)
+            )
+            if self.verbose:
+                print_text(f"After Reranking, new rank list for nodes: {rerank_ranks}")
+
+            initial_results: List[NodeWithScore] = []
+
+            for idx in rerank_ranks:
+                initial_results.append(
+                    NodeWithScore(node=nodes[idx].node, score=nodes[idx].score)
+                )
+            return initial_results[: self.top_n]
+        else:
+            return nodes[: self.top_n]
```

Comment:
nit: can probably define a helper method here and use it in both `sync` and `async` to reduce repetition.